src/plugins/syscalls/win.cpp

```diff
@@ -175,7 +175,7 @@ static event_response_t syscall_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
             if ( VMI_FAILURE == vmi_read(vmi, &ctx, s->reg_size * nargs, &tmp_args[0], NULL) )
                 nargs = 0;
 
-            for (auto const& i: tmp_args)
+            for (size_t i = 0; i < nargs; ++i)
                 args[i] = tmp_args[i];
```

Comment:
I mean you could just write `args = tmp_args` :thinking:

Reply:
@BonusPlay, would it convert `uint32_t` to `uint64_t`?

Reply:
You're right, I didn't notice `32` vs `64`.

Reply:
```
syscalls/win.cpp:178:18: error: no viable overloaded '='
            args = tmp_args;
            ~~~~ ^ ~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/stl_vector.h:447:7: note: candidate function not viable: no known conversion from
      'vector<uint32_t, allocator<unsigned int>>' to 'const vector<unsigned long, allocator<unsigned long>>' for 1st argument
      operator=(const vector& __x);
```

Reply:
Oh, btw: it wasn't "out of bounds access". It was just confusing
```cpp
for (auto const& i: tmp_args)
    args[i] = tmp_args[i];
```
you iterate over elements, and then use an element as index?

```cpp
for (auto const& i: tmp_args)
    args.push_back(i);
```
This is safe and doesn't include any indexes :)

Reply:
I have thought about this. But in this case we would have to make something like this:
```c++
    std::vector<uint64_t> args; // previous: std::vector<uint64_t> args(nargs);

    if ( 4 == s->reg_size ) {
        for (auto const& i: tmp_args)
            args.push_back(i);
    } else {
        args.resize(nargs);
        vmi_read(vmi, &ctx, sp_size, &args[4], NULL)
    }
```